Test/TestHelper.cs

```diff
@@ -77,25 +79,49 @@ internal static List<RawCapture> RunCapture(string filter, Action<PcapDevice> ro
             var device = GetPcapDevice();
             Console.WriteLine($"Using device {device}");
             var received = new List<RawCapture>();
-            device.Open(DeviceMode.Normal, 1);
+            var mode = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ?
+                DeviceMode.Promiscuous :
+                DeviceMode.Normal;
+
+            device.Open(mode, 1);
             device.Filter = filter;
+            // We can't use the same device for capturing and sending in Linux
```

Comment:
why can't we?

Reply:
The device simply does not receive its own sent traffic in Linux, not sure what the reason is.

Tested using Linux on Windows WSL

Reply:
@kayoub5 would you mind rebasing that commit and adding the 'why' to the comment so we have a record of the reason?

Reply:
Done